### PR TITLE
Dev 487 new funding design for bounty page

### PR DESCRIPTION
**Pull request by vhcsilva**

### CODE REVIEW

File: components/bounty/funding-section/fund-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  const [isExecuting, setIsExecuting] = useState(false);
+
+  const { handleFundBounty } = useBepro();
+  const { dispatch } = useContext(ApplicationContext);
+  const { allowance, balance, setAddress, approve } = useERC20();
+  const { activeIssue, networkIssue, getNetworkIssue } = useIssue();
+
+  const bountyId = activeIssue?.contractId || networkIssue?.id || "XX";
+  const fundBtnDisabled = 
+    isExecuting || !amountToFund || amountToFund + networkIssue?.fundedAmount > networkIssue?.fundingAmount;
```
Comment (moshmage) [inline components/bounty/funding-section/fund-modal.tsx:40]: shouldn't this be `>=` instead ?

Comment (vhcsilva) [reply components/bounty/funding-section/fund-modal.tsx:40]: It verify if the value entered by the user is greater than the available amount to fund.
If a funding request is asking for 100 BEPRO and already received 50 BEPRO, new funders can't fund more than 50 BEPRO.

File: components/issue-proposal-progress-bar.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export default function IssueProposalProgressBar() {
   const isFinalized = !!networkIssue?.closed;
   const isInValidation = !!networkIssue?.isInValidation;
   const isIssueinDraft = !!networkIssue?.isDraft;
-  const creationDate = networkIssue?.creationDate;
+  const creationDate = networkIssue?.creationDate || new Date();
```
Comment (moshmage) [inline components/issue-proposal-progress-bar.tsx:31]: it's a bit impossible for creationDate to not exist, but if it doesnt: It shouldn't be "todays date"

**Review by moshmage — CHANGES_REQUESTED** [review]

File: services/dao-service.ts [hunk 3]
```diff
@@ -386,6 +386,14 @@ export default class DAO {
                                    githubUser);
   }
 
+  async fundBounty(bountyId: number, amount: number, tokenDecimals?: number): Promise<TransactionReceipt> {
```
[the remark this answers is not in the document]
Comment (vhcsilva) [reply services/dao-service.ts:389]: @moshmage I think every interaction with the contract should be through the `dao-service`, so we can manage these interactions and refactor the code easily.

File: pages/[network]/bounty.tsx [hunk 4]
```diff
@@ -102,7 +105,7 @@ export default function PageIssue() {
           </CustomContainer>
       }
 
-      {networkIssue ? (
+      { wallet?.address ? (
```
[the remark this answers is not in the document]
Comment (vhcsilva) [reply pages/[network]/bounty.tsx:108]: @moshmage This condition is used to show or hide the bounty roadmap. Should I change to show the  roadmap even if there is no wallet connected ?

[the remark this answers is not in the document]
Comment (moshmage) [reply pages/[network]/bounty.tsx:108]: `IssueDescription` seems to be inside this condition as well ?

**Pull request merged**